CMSgov/onemac
# OY2-10904: migrate browser confirm to custom modal dialog

Story: https://qmacbis.atlassian.net/browse/OY2-10904
Endpoint: https://dn2cg7tgl1boi.cloudfront.net/

### Details

Replaces `window.confirm` usages with a confirmation dialog based on `@cmsgov/design-system`'s `Dialog` component.

### Changes

- All instances of `confirm(...)` were updated to this new component
- The State User signup pages now have a confirmation on cancel (they did not previously)
- Language in the dialogs has been updated per guidance from HCD

### Implementation Notes

- Added a new component named `ConfirmationDialog`, a wrapper around the `Dialog` component with some inbuilt assumptions and default props. Usage should (mostly) be obvious given the docs for the latter.

### Test Plan

1. Log in as `stateadminactiveMI` and go to the User Management page. Check the flows for Grant, Deny, and Revoke for bugs and language.
2. Go to the Dashboard and select a submission type. Once you are in the submission form, click the back button in our header. Make sure that the confirmation dialog works and makes sense.
3. Click the Cancel button at the bottom of the form. Repeat the sanity check for both action buttons and clickaway.
4. Log out, then back in as `cmsreviewerunregistered`. Select CMS Reviewer on the role selection page, then check the confirmation flow as before for the Cancel button underneath the dropdowns.
5. Repeat step 4 with `statesubmitterunregistered`.
